netket/optimizer/qgt/qgt_jacobian_pytree.py

```diff
@@ -19,24 +19,45 @@
 from jax import numpy as jnp
 from flax import struct
 
-from netket.utils.types import PyTree
+from netket.utils.types import PyTree, Array
 from netket.utils import n_nodes
 from netket.stats import sum_inplace
 import netket.jax as nkjax
 
 from ..linear_operator import LinearOperator, Uninitialized
 
-from .treemv_logic import mat_vec
+from .qgt_jacobian_pytree_logic import mat_vec, prepare_doks
 
 
-def QGTJacobianPyTree(vstate, *, mode, rescale_shift=False) -> "QGTJacobianT":
-    O, scale = 0, 0  # compute gradient
+def QGTJacobianPyTree(
+    vstate, *, mode="auto", rescale_shift=True, **kwargs
```

Comment:
this is what was breaking it.
Do we want it on by default?

```suggestion
    vstate, *, mode="auto", rescale_shift=False, **kwargs
```

Reply:
in my experience, rescale_shift is best kept as optional... it tends to be unstable